compiler/modules/comm.diagnostic/diagnostic.bal

```diff
@@ -112,7 +114,40 @@ public function messageFormat(Template t) returns string {
 
 public function format(Diagnostic d) returns string[] {
     Location loc = d.location;
+    string[] lines = [];
+    string line = loc.file.filename();
     LineColumn lc = locationLineColumn(loc);
     // lineColumn returns 0-based lines currently
-    return [loc.file.filename() + ":" + lc[0].toString() + ":" + (lc[1] + 1).toString() + ": error: " + d.message];
+    line += ":" + lc[0].toString() + ":" + (lc[1] + 1).toString();
+    line += ": error: " + d.message;
+    lines.push(line);
+    Range|Position range = loc.range;
+    lines.push(range is Position ? loc.file.lineContent(range) : loc.file.lineContent(range.startPos));
+    lines.push(caretLine(loc.file, range));
+    return lines;
+}
+
+function caretLine(File file, Range|Position range) returns string {
+    string padding;
+    string carets = "^";
+    if range is Position {
```

Comment:
Compute the startColumn inside the if/else. Then compute the padding once at the end in the return statement.

Reply:
fixed